edc-tests/edc-dataplane/cloud-transfer-tests/src/test/java/org/eclipse/tractusx/edc/dataplane/transfer/test/AzureToAzureTest.java

```diff
@@ -93,6 +97,43 @@ void setup() {
         consumerBlobHelper = new AzureBlobHelper(AZBLOB_CONSUMER_ACCOUNT_NAME, AZBLOB_CONSUMER_ACCOUNT_KEY, azuriteContainer.getHost(), azuriteContainer.getMappedPort(AZURITE_CONTAINER_PORT));
     }
 
+    @Test 
+    void transferMultipleFile_success() {
+        var sourceContainer = providerBlobHelper.createContainer(AZBLOB_PROVIDER_CONTAINER_NAME);
+        var filesNames = new ArrayDeque<String>();
+        for (var i : IntStream.rangeClosed(1, 5).toArray()) {
+            var complement = new StringBuilder().append(PREFIX_FOR_MUTIPLE_FILES).append(i).append('_');
+            filesNames.add(complement.toString() + "" + TESTFILE_NAME);
+            providerBlobHelper.uploadBlobWithPrefix(sourceContainer, filesNames.getLast(), complement.toString());
+        }
```

Comment:
this piece looks unnecessarily complex, and the `uploadBlobWithPrefix` has a signature that's not really easy to understand.
My suggestion would be to, instead of introduce that new method, to add a new version of the `uploadBlob` that could get a `BinaryData` object like:
```java
    public void uploadBlob(BlobContainerClient client, BinaryData data, String targetBlobName) {
        client.getBlobClient(targetBlobName).upload(data, true);
    }
```

this way you won't need to deal with prefixes and stuff, you'll just read the `TESTFILE_NAME` once in a `BinaryData` variable like
```java
 var fileData = BinaryData.fromString(TestUtils.getResourceFileContentAsString(TESTFILE_NAME));
```

and you could separate the "fileNames build" and the upload in 2 different statements like:
```java
var fileNames = IntStream.rangeClosed(1, 5).mapToObj(i -> PREFIX_FOR_MUTIPLE_FILES + i + '_' + TESTFILE_NAME).toList();
        
fileNames.forEach(filename -> providerBlobHelper.uploadBlob(sourceContainer, fileData, filename));
```

this is valid for all the "multiple file" tests.
At the end you could also get rid of all the other `uploadBlob` methods, because they could all be collapsed to the version that accepts the `BinaryData`

Reply:
yes, great suggestion 👌